theme/helpers.js

```diff
@@ -9,13 +13,81 @@ exports.Color = class {
         Object.keys(shades).forEach((shade) => (this[shade] = shades[shade]));
     }
 
-    shade(shade) {
+    alter(shade = 500) {
 
-        return this[shade];
+        return Color(this[shade]);
     }
 
     toString() {
 
         return this['500'];
     }
 };
+
+exports.paletteFromTheme = (theme) => {
+
+    const colors = Object.keys(theme).reduce((acc, key) => {
+
+        const regex = /^color-(?<color>\w+)-(?<transparent>\w+-)?(?<shade>\d{3})/;
+
+        Assert(regex.test(key), `Expected ${key} to match regex ${regex}`);
+
+        const { color, transparent, shade } = key.match(/^color-(?<color>\w+)-(?<transparent>\w+-)?(?<shade>\d{3})/).groups;
+
+        if (!(color in acc)) {
+            acc[color] = { transparent: {} };
+        }
+
+        if (transparent) {
```

Comment:
Would `color-basic-transparent-100` and `color-basic-control-transparent-100` both set `acc.basic.transparent[100]`?